src/tests/units/operator_tests.cpp

```diff
@@ -66,6 +70,33 @@ void tests(DeviceType device, Type floatType = Type::float32) {
     CHECK(compare(rmin2,  [](float a) {return std::min(1.f, a);}));
   }
 
+  SECTION("Max all negative. <= 32 Elements") {
+    graph->clear();
+    values.clear();
+
+    std::vector<T> vA({-1, -2, -3, -4});
+    auto a = graph->constant({1, 1, 4}, inits::fromVector(vA));
+
+    auto compare = [&](Expr res, std::function<float(float, float)> op) -> bool {
+      if (res->shape().elements() != 1)
+          return false;
+      float val = res->val()->get(0);
+      
+      float reduced = vA[0];
+      for(const auto val : vA) {
+        reduced = op(reduced, val);
+      }
+      return floatApprox(reduced, val);
+    };
+
+    // @TODO: add all operators here for completeness
+    auto maxReduce = max(a, -1);
```

Comment:
Could you add /*axis=*/ before -1?
And, it would be great to add a few more reduction operators.

Reply:
Done